### PR TITLE
fix: use explicit session in `PaLM2TextGenerator`

### PR DESCRIPTION
**Pull request by shobsi**
This is to avoid running into conflict with the global session which may have different options (e.g. location, connection etc.) set.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Review by tswast — APPROVED** [review]

**Review by ashleyxuu — APPROVED** [review]

File: bigframes/ml/llm.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def _from_bq(
         kwargs: dict = {}
         last_fitting = model.training_runs[-1]["trainingOptions"]
 
-        dummy_text_generator = cls()
+        dummy_text_generator = cls(session=session)
```
Comment (GarrettWu) [inline bigframes/ml/llm.py:147]: I don't think this is the issue. It is only a dummy object to retrieve param types. The actual creation is in line 155.

**Review by GarrettWu — CHANGES_REQUESTED** [review]

File: bigframes/ml/llm.py (hunk 1 above)
Comment (shobsi) [reply bigframes/ml/llm.py:147]: This does call the `.ctor` -> `_create_bqml_model` -> `create_bq_connection`?

Comment (GarrettWu) [reply bigframes/ml/llm.py:147]: I see. Thanks for figuring out.

**Review by GarrettWu — APPROVED** [review]

File: bigframes/ml/llm.py (hunk 1 above)
Comment (GarrettWu) [reply bigframes/ml/llm.py:147]: Actually this whole block is unnecessary and does more than what we want. I'll send another PR to refactor some.

**Pull request merged**